### PR TITLE
Add architecture diagram.

### PR DESCRIPTION
**Pull request by rmmh**
/cc @kubernetes/sig-testing-pr-reviews

### CODE REVIEW

File: docs/architecture.svg [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+<!-- Mungegithub&#45;&gt;Splice -->
+<g id="edge4" class="edge"><title>Mungegithub&#45;&gt;Splice</title>
+<path fill="none" stroke="black" d="M610,-816.21C610,-788.417 610,-742.594 610,-713.917"/>
+<polygon fill="black" stroke="black" points="606.5,-816.243 610,-826.243 613.5,-816.243 606.5,-816.243"/>
+<text text-anchor="middle" x="626.5" y="-785.8" font-family="Times,serif" font-size="14.00">read</text>
+<text text-anchor="middle" x="626.5" y="-770.8" font-family="Times,serif" font-size="14.00">queue</text>
+</g>
+<!-- ApiServer -->
+<g id="node19" class="node"><title>ApiServer</title>
+<polygon fill="none" stroke="black" points="543.325,-565.403 531.179,-590.485 482,-601.648 432.821,-590.485 420.675,-565.403 454.708,-545.288 509.292,-545.288 543.325,-565.403"/>
```
Comment (ixdy) [inline docs/architecture.svg]: heptagon is a nice attention to detail. :)

**Comment by spxtr** [comment]
This looks awesome. You didn't include `bootstrap.py`, but that's fine I guess.

**Comment by fejta** [comment]
s/Job/Job pod? Also maybe move it out of the blue box (and clarify what defines the blue box)
/lgtm

File: docs/architecture.dot [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		Hook -> ApiServer [label="create ProwJob"]
+		ApiServer -> Job [label="start Pod"]
+		ApiServer -> Sinker [dir="back"]
+		ApiServer -> Horologium [dir="back"]
+	}
+
+	GCS -> PubSub
+	Kettle -> GCS [label="read"]
+	Kettle -> PubSub [label="read"]
+	Kettle -> BigQuery [label="write"]
```
Comment (spiffxp) [inline docs/architecture.dot]: The arrow direction confuses me.  I'm expecting it to indicate the flow of data, in which case I'd expect the "read" arrows to go the other way.  OTOH, if it's supposed to imply "depends on / won't function without" then I think? other arrows are in the wrong direction, like testgrid->gcs or github->hook

**Comment by spiffxp** [comment]
this is awesome

File: docs/architecture.dot (hunk 2 above)
Comment (rmmh) [inline docs/architecture.dot]: It's a layout thing. I added a note about how dir=back works.

The arrows indicate RPC directions. Github POSTs events to Hook's HTTP endpoint.

**Review by Cirruslucent — COMMENTED** [review]
Amazing diagram. Containerized applications would totally change the way distribution is ran

**Pull request merged**